rskj-core/src/main/java/org/ethereum/core/BlockFactory.java

```diff
@@ -154,9 +154,20 @@ private BlockHeader decodeHeader(RLPList rlpHeader, boolean sealed) {
             ummRoot = rlpHeader.get(r++).getRLPRawData();
         }
 
+        byte version = 0x0;
+
+        if (activationConfig.isActive(ConsensusRule.RSKIP351, blockNumber)) {
```

Comment:
I think we discussed not using the blocknumber for detecting the version, but using the length of the field itself.

Reply:
Anyway, don't think this is a blocker

Reply:
The implementation has the property of allowing parsing the blockchain with the length and without the activation. I think this is a more clear hard fork detection anyway